domainobjs/ts/__tests__/stateLeaf.test.ts

```diff
@@ -2,54 +2,141 @@ import { expect } from "chai";
+      const randomLeaf2 = StateLeaf.genRandomLeaf();
+      expect(randomLeaf.equals(randomLeaf2)).to.eq(false);
+    });
+  });
+
+  describe("equals", () => {
+    it("should return true when comparing two equal state leaves", () => {
+      const stateLeaf = new StateLeaf(pubKey, BigInt(123), BigInt(1231267));
+
+      const stateLeaf2 = new StateLeaf(pubKey, BigInt(123), BigInt(1231267));
 
-    const arr = stateLeaf.asCircuitInputs();
-    expect(arr).to.be.instanceOf(Array);
-    expect(arr.length).to.be.gt(0);
-    expect(arr.length).to.eq(4);
+      expect(stateLeaf.equals(stateLeaf2)).to.eq(true);
+    });
+
+    it("should return false when comparing two different state leaves", () => {
+      const stateLeaf = new StateLeaf(pubKey, BigInt(123), BigInt(1231267));
+
+      const stateLeaf2 = new StateLeaf(pubKey, BigInt(123), BigInt(1231268));
+
+      expect(stateLeaf.equals(stateLeaf2)).to.eq(false);
+    });
   });
 
-  it("equals should return true when comparing two equal state leaves", () => {
-    const stateLeaf = new StateLeaf(pubKey, BigInt(123), BigInt(1231267));
+  describe("serialization", () => {
+    describe("serialize", () => {
+      it("should work correctly", () => {
+        const stateLeaf = new StateLeaf(pubKey, BigInt(123), BigInt(1231267));
+
+        const serialized = stateLeaf.serialize();
+        expect(serialized).to.not.eq(null);
+      });
+    });
+
+    describe("deserialize", () => {
+      it("should work correctly", () => {
+        const stateLeaf = new StateLeaf(pubKey, BigInt(123), BigInt(1231267));
+
+        const serialized = stateLeaf.serialize();
+        const deserialized = StateLeaf.deserialize(serialized);
+        expect(deserialized.equals(stateLeaf)).to.eq(true);
+      });
+    });
+
+    describe("toJSON", () => {
+      it("should produce an object with the correct properities", () => {
+        const stateLeaf = new StateLeaf(pubKey, BigInt(123), BigInt(1231267));
+
+        const json = stateLeaf.toJSON();
+        expect(json).to.not.eq(null);
+
+        expect(Object.keys(json)).to.deep.eq(["pubKey", "voiceCreditBalance", "timestamp"]);
+      });
+    });
+
+    describe("fromJSON", () => {
+      it("should produce a state leaf from a JSON object", () => {
+        const stateLeaf = new StateLeaf(pubKey, BigInt(123), BigInt(1231267));
+
+        const json = stateLeaf.toJSON();
+        const deserialized = StateLeaf.fromJSON(json);
+        expect(deserialized.equals(stateLeaf)).to.eq(true);
+      });
+    });
+  });
 
-    const stateLeaf2 = new StateLeaf(pubKey, BigInt(123), BigInt(1231267));
+  describe("asCircuitInputs", () => {
+    it("should return an array", () => {
+      const stateLeaf = new StateLeaf(pubKey, BigInt(123), BigInt(1231267));
 
-    expect(stateLeaf.equals(stateLeaf2)).to.eq(true);
+      const arr = stateLeaf.asCircuitInputs();
+      expect(arr).to.be.instanceOf(Array);
+      expect(arr.length).to.be.gt(0);
+      expect(arr.length).to.eq(4);
```

Comment:
Both of them are checking the `arr.length`, should we just check the if the `arr.length` equal to 4?